common/management/src/user/user_mgr.rs

```diff
@@ -80,10 +87,18 @@ impl UserMgrApi for UserMgr {
         }
     }
 
-    async fn get_users(&self) -> Result<Vec<SeqV<UserInfo>>> {
-        let user_prefix = self.user_prefix.clone();
+    async fn get_users(&self, username: Option<String>) -> Result<Vec<SeqV<UserInfo>>> {
```

Comment:
get_users aims to get all the users prefix a tenant, why we add a username here?

Reply:
`get_users` is for a tenant to get all his users.
In databend, the first level is a tenant, then a tenant may has many users for queries.

Reply:
> `get_users` is for a tenant to get all his users.

Yes, we can get all users under a tenant through `get_users(None)`.

Or, we need to separate `get_users` and `get_users_by_name` ?

Reply:
Thanks for explaintion.
`get_users` add the `Option` is for test only? Since we have a `get_user` api with username already.

Reply:
Sorry, I think I might not describe it clearly.

This pr use both 'user_name'@'host_name' as the key, `get_user` aims to get one user by username and hostname, `get_users` for list users by username, get all user if username is None.

There are two scenarios for `get_users`:
1. list all user for a tenant, as as `show users`;
2. list users by filter username to check client address in login. For example, the username has two hostnames `localhost` and `%`, we should check whether the user's client address matches the hostname.

Reply:
>list all user for a tenant, as as show users;


tenant is configed in the query/config already, and init in:
https://github.com/datafuselabs/databend/blob/0d5fc199c0834aa11ccb2b115d561822d4df208b/common/management/src/user/user_mgr.rs#L42
`get_users()` defaults to get all the users under the tenant

> list users by filter username to check client address in login. For example, the username has two hostnames localhost and 192.168.0.0/16, we should check whether the user's client address matches the hostname.


We should check : <username, hostname> togther, no need to change the `get_users` api to get users who prefix with username first then check the hostname,  change this api looks ok:
```
get_user(&self, user: &str, hostname: &str) -> Result<UserInfo>
```
to
```
get_user(&self, user: &str, hostname: &str) -> Result<Option<UserInfo>>
```

If it is None -- No users+hostname found.

Reply:
> On balance, we don't need to modify the get_users api.

I will revoke the changes to get_users first.

Reply:
> We should check : <username, hostname> togther, no need to change the get_users api to get users who prefix with username first then check the hostname, get_user method works for your case

I have a confusion, if the user's client address is `192.168.10.100`, how to match the above two hostnames `localhost` and `192.168.0.0/16` through get_user.

Reply:
Aha, finally got your points, the host name can take many forms, and wildcards! Thank you @Junnplus 

The get_user api change looks make sense

Reply:
> Aha, finally got your points, the host name can take many forms, and wildcards!
😅 

> But we can wild them in the get_user(First, get the users prefixed the username, and regex the hostname from the users) api and not changed the get_users api.

Or we can add get_prefix_users instead of get_user for filter username.

Reply:
That's 👌

Reply:
Thanks @BohuTANG , We can seperate a PR to support authentication.

Are there any other questions for this PR?